howso/engine/session.py

```diff
@@ -184,22 +190,22 @@ def set_metadata(self, metadata: Optional[Dict[str, Any]]) -> None:
         self._metadata = metadata
         self._update()
 
-    def _update_attributes(self, session: BaseSession) -> None:
+    def _update_attributes(self, session) -> None:
         """
         Update the protected attributes of the session.
 
         Parameters
         ----------
-        session : BaseSession
+        session : Session
             The base session instance.
 
         Returns
         -------
         None
         """
-        for key in self.attribute_map.keys():
+        for key in Session.attribute_map.keys():
             # Update the protected attributes directly since the values
-            # have already been validated by the "BaseSession" instance
+            # have already been validated by the "Session" instance
             # and to prevent triggering an API update call
             setattr(self, f'_{key}', getattr(session, key))
```

Comment:
```suggestion
            setattr(self, f'_{key}', session.get(key))
```

Reply:
Should be addressed by: d66654ea10038f70a275bb4aa504128552745e60

Reply:
Changes still missing in latest commit

Reply:
Sorry for missing that. Addressed by:  eab8647